pkg/chart/loader/archive.go

```diff
@@ -85,7 +85,10 @@ func ensureArchive(name string, raw *os.File) error {
 	if err != nil && err != io.EOF {
 		return fmt.Errorf("file '%s' cannot be read: %s", name, err)
 	}
-	if contentType := http.DetectContentType(buffer); contentType != "application/x-gzip" {
+
+	// Helm may identify achieve of the application/x-gzip as application/vnd.ms-fontobject.
+	// Fix for: https://github.com/helm/helm/issues/12261
+	if contentType := http.DetectContentType(buffer); contentType != "application/x-gzip" && !isGZipApplication(buffer) {
```

Comment:
Have you looked to see if this is an upstream bug? Why is [DetectContentType](https://pkg.go.dev/net/http#DetectContentType) returning vnd.ms-fontobject?

Reply:
This is not an upstream bug. I occasionally encounter this bug. The achieve I use meets both the `application/x-gzip` and `application/vnd.ms-fontobject` types. However, because `application/vnd.ms-fontobject` matches first, so `DetectContentType` returning vnd.ms-fontobject

Here's the questionable achieve I used, which can be decompressed and tested.
[rediscluster-24.1.110000008.tgz.zip](https://github.com/helm/helm/files/12299544/rediscluster-24.1.110000008.tgz.zip)

Reply:
@joejulian -- `DetectContentType` has a bunch of heuristics:

https://github.com/helm/helm/issues/12261#issuecomment-1661247878

Reply:
Thanks!

Reply:
zip != gzip, now the .tgz inside the zip file : rediscluster-24.1.110000008.tgz would be detected correctly as application/x-gzip